src/main/java/org/datadog/jmxfetch/App.java

```diff
@@ -290,6 +291,24 @@ public TaskStatusHandler invoke(
         }
     }
 
+    /**
+     * Builds an {@link ExecutorService} of the specified fixed size. Threads will be created
+     * and executed as daemons if {@link AppConfig#isDaemon()} is true. Defaults to false.
+     *
+     * @param size The thread pool size
+     * @return The create executor
+     */
+    private ExecutorService buildExecutorService(int size) {
+        return Executors.newFixedThreadPool(size, new ThreadFactory() {
+            @Override
+            public Thread newThread(Runnable runnable) {
+                Thread thread = Executors.defaultThreadFactory().newThread(runnable);
```

Comment:
I would suggest assigning a proper name to the thread... this would have greatly helped our troubleshooting.  Maybe next time someone does some work here?